docs/proposals/20220411-cluster-api-state-metrics.md

```diff
@@ -0,0 +1,359 @@
+      - [Scrapable Information](#scrapable-information)
+      - [Relationship to kube-state-metrics](#relationship-to-kube-state-metrics)
+      - [Reusage of kube-state-metrics packages](#reusage-of-kube-state-metrics-packages)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+    - [Directly use kube-state-metrics](#directly-use-kube-state-metrics)
+    - [Expose metrics by the controllers](#expose-metrics-by-the-controllers)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Metrics](#metrics)
+      - [Cluster CR](#cluster-cr)
+      - [KubeadmControlPlane CR](#kubeadmcontrolplane-cr)
+      - [MachineDeployment CR](#machinedeployment-cr)
+      - [MachineSet CR](#machineset-cr)
+      - [Machine CR](#machine-cr)
+      - [MachineHealthCheck CR](#machinehealthcheck-cr)
+    - [Gaduation Criteria](#gaduation-criteria)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+- `CustomResource` (CR)
+- `CustomResourceDefinition` (CRD)
+- Cluster API State Metrics (CASM)
+
+Refer to the [Cluster API Book Glossary].
+
+## Summary
+
+This proposal outlines a new experimental component for exposing metrics specific to CAPI's CRs.
+
+This component is derived from  [mercedes-benz/cluster-api-state-metrics] and all the merits for its inception goes to the team that created it: [@chrischdi](https://github.com/chrischdi), [@seanschneeweiss](https://github.com/seanschneeweiss), [@tobiasgiese](https://github.com/tobiasgiese).
+
+A special thank goes to [Mercedes Benz] who allowed the donation of this project to CNCF and to the Cluster API community.
+
+## Motivation
+
+As of now CAPI controllers only expose metrics that are provided by controller-runtime, which in turn are specific to the controllers internal behavior and thus do not provide any information about the state of the clusters provisioned by CAPI.
+
+This proposal introduces a new component that can generate a new set of metrics that will help end users to monitor the state of the Cluster API resources in their [OpenMetrics] compatible monitoring system of choice.
+
+### Goals
+
+- Define metrics to monitor the state of the Cluster API resources, thus solving parts of the [metrics umbrella issue #1477]
+- Implementing a new component, which exposes the above metrics by following the [OpenMetrics] standard
+- Make metrics part of the CAPI developer workflow by making them accessible via Tilt and possible also as a E2E test artifacts
+- Implement metrics for core CAPI CRs
+
+### Non-Goals
+
+- Implement metrics for provider specific CAPI CRs
+
+### Future Work
+
+- Metrics for resources of Cluster API providers (e.g., CAPA, CAPO, CAPZ, ...)
+- Collection of example alerts to e.g. alert when
+  - a KubeadmControlPlane is not healthy.
+  - 70% of my worker Machines are not healthy.
+  - 70% of my Machines are blocked on deletion.
+
+## Proposal
+
```

Comment:
All this section does a great job explaining what will be scrapped and why but can we get into the impl details of how cluster-api-state-metrics will work? diagrams are useful as well.

Reply:
Kindly see other comment and additional commit :-)